integrations/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/NessieUtils.scala

```diff
@@ -167,32 +162,104 @@ object NessieUtils {
       currentCatalog: CatalogPlugin,
       catalog: Option[String]
   ): NessieApiV1 = {
+    val maybeIcebergCatalog = getBaseIcebergCatalog(currentCatalog, catalog)
+    val errorPre =
+      "The command works only when the catalog is a NessieCatalog or a RESTCatalog using the Nessie Catalog Server"
+    val errorPost =
+      "Either set the catalog via USE <catalog_name> or provide the catalog during execution: <command> IN <catalog_name>."
+    require(maybeIcebergCatalog.isDefined, errorPre + ". " + errorPost)
+
+    val icebergCatalog = maybeIcebergCatalog.get
+
     val catalogName = catalog.getOrElse(currentCatalog.name)
-    val sparkConf = SparkSession.active.sparkContext.conf
-    val catalogConf = sparkConf
-      .getAllWithPrefix(s"spark.sql.catalog.$catalogName.")
-      .toMap
 
-    val catalogImpl = catalogConf.get("catalog-impl")
-    val catalogErrorDetail = catalogImpl match {
-      case Some(clazz) => s"but $catalogName is a $clazz"
-      case None =>
-        s"but spark.sql.catalog.$catalogName.catalog-impl is not set"
-    }
-    // Referring to https://github.com/apache/iceberg/blob/master/nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java
-    // Not using fully-qualified class name to provide protection from shading activities (if any)
-    require(
-      catalogImpl
-        .exists(impl => impl.endsWith(".NessieCatalog")),
-      s"The command works only when the catalog is a NessieCatalog ($catalogErrorDetail). Either set the catalog via USE <catalog_name> or provide the catalog during execution: <command> IN <catalog_name>."
-    )
+    val nessieClientConfigMapper: java.util.function.Function[String, String] =
+      icebergCatalog.getClass.getSimpleName match {
+        case "NessieCatalog" =>
+          val sparkConf = SparkSession.active.sparkContext.conf
+          val catalogConf = sparkConf
+            .getAllWithPrefix(s"spark.sql.catalog.$catalogName.")
+            .toMap
+          x => catalogConf.getOrElse(x.replace("nessie.", ""), null)
+        case "RESTCatalog" =>
+          val catalogProperties = getCatalogProperties(icebergCatalog)
+
+          require(
+            java.lang.Boolean
+              .parseBoolean(catalogProperties.get("nessie.is-nessie-catalog")),
+            errorPre + ", but the referenced REST endpoint is not a Nessie Catalog Server. " + errorPost
+          )
+          x => {
+            x match {
+              case NessieConfigConstants.CONF_NESSIE_URI =>
+                // Use the Nessie Core REST API URL provided by Nessie Catalog Server. The Nessie Catalog
+                // Server provides a _base_ URI without the `v1` or `v2` suffixes. We can safely assume
+                // that `nessie.core-base-uri` contains a `/` terminated URI.
+                catalogProperties.get("nessie.core-base-uri") + "v1"
```

Comment:
It's for the Spark SQL extensions - there were some issues with those using v2 (to be tackled in a different change).
Since the v1/v2 decision is overall only present in the Nessie Spark SQL extensions, it's fine for now.